cardano-node/test/Test/Common/Process.hs

```diff
@@ -0,0 +1,36 @@
+module Test.Common.Process
+  ( createProcess
+  , interruptProcessGroupOf
+  , waitForProcess
+  ) where
+
+import           Control.Monad.IO.Class
+import           Data.Function ((.))
+import           Data.Function (($))
+import           Data.Maybe (Maybe (..))
+import           GHC.Stack (HasCallStack)
+import           System.Exit (ExitCode)
+import           System.IO (Handle, IO)
+import           System.Process (CreateProcess (..), ProcessHandle)
+
+import qualified GHC.Stack as GHC
+import qualified Hedgehog as H
+import qualified System.Process as IO
+
+createProcess :: HasCallStack
+  => CreateProcess
+  -> H.PropertyT IO (Maybe Handle, Maybe Handle, Maybe Handle, ProcessHandle)
+createProcess cp = GHC.withFrozenCallStack $ do
+  H.evalM . liftIO $ IO.createProcess cp
+
+interruptProcessGroupOf :: HasCallStack
+  => ProcessHandle
+  -> H.PropertyT IO ()
+interruptProcessGroupOf hProcess = GHC.withFrozenCallStack $ do
+  H.evalM . liftIO $ IO.interruptProcessGroupOf hProcess
+
+waitForProcess :: HasCallStack
+  => ProcessHandle
+  -> H.PropertyT IO ExitCode
+waitForProcess hProcess = GHC.withFrozenCallStack $ do
+  H.evalM . liftIO $ IO.waitForProcess hProcess
```

Comment:
Maybe `waitForProcess` is indefinitely waiting?

Reply:
I thought of that possibility, but there's no visibility in CI, so how can I confirm?

Reply:
I'm also wondering if there is something spinning now on the build server.  Usually there should be a timeout on builds if there is no progress.

Reply:
You will need to get in touch with devops for assistance.